src/daft-session/src/options.rs

```diff
@@ -1,10 +1,39 @@
+use daft_catalog::FindMode;
+
 // TODO make env variables
 pub(crate) const _DAFT_SESSION: &str = "default";
 pub(crate) const _DAFT_SESSION_USER: &str = "daft";
 pub(crate) const _DAFT_SESSION_TEMP_DIR: &str = "/tmp";
 
+/// Session state variables.
 #[derive(Debug, Default)]
 pub(crate) struct Options {
+    pub identifier_mode: IdentifierMode,
     pub curr_catalog: Option<String>,
     pub curr_namespace: Option<Vec<String>>,
 }
+
+/// Identifier mode controls identifier resolution and name binding logic (tables, columns, views, etc).
+#[derive(Debug, Default)]
+#[allow(dead_code)]
+pub enum IdentifierMode {
+    /// Find case-insensitive (rvalue) and bind case-preserved (lvalue).
+    Insensitive,
+    /// Find case-sensitive (rvalue) and bind case-preserved (lvalue).
+    #[default]
+    Sensitive,
+    /// Find case-sensitive (rvalue) and bind (lower) case-normalized (lvalue).
+    Normalize,
```

Comment:
Consider adding a simple example for each identifier mode to make the documentation more approachable. The current terminology (`rvalue`/`lvalue`) is specialized and might not be immediately clear to all contributors.